### PR TITLE
umoria: unpin from gcc-9

### PR DESCRIPTION
**Pull request by trofi**
Needed a small upstream patch to fix the build.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by trofi** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 351467`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>umoria</li>
  </ul>
</details>

**Review by aciceri — APPROVED** [review]

File: pkgs/games/umoria/default.nix [hunk 1]
```diff
@@ -21,6 +22,15 @@ gcc9Stdenv.mkDerivation rec {
     sha256 = "sha256-1j4QkE33UcTzM06qAjk1/PyK5uNA7E/kyDe3bZcFKUM=";
   };
 
+  patches = [
```
Comment (Artturin) [inline pkgs/games/umoria/default.nix:25]: https://github.com/dungeons-of-moria/umoria/commit/f294e5880cd21d25c11eee820d629f4ff504ad10 this commit will restore darwin support
Should apply cleanly

Comment (trofi) [reply pkgs/games/umoria/default.nix:25]: Applied, and a few more hacks on top to get `pkgsLLVM.umoria` build on `linux`.

**Comment by trofi** [comment]
`clang` build will also need a fix similar to https://github.com/dungeons-of-moria/umoria/pull/87.

I'll draft the PR until upstream considers the fix.

**Comment by trofi** [comment]
> `clang` build will also need a fix similar to [dungeons-of-moria/umoria#87](https://github.com/dungeons-of-moria/umoria/pull/87).
> 
> I'll draft the PR until upstream considers the fix.

Declaring timeout. Pluued patch pending upstream, inclusion. Ready for review.

File: pkgs/by-name/um/umoria/package.nix [hunk 2]
```diff
@@ -21,6 +22,38 @@ gcc9Stdenv.mkDerivation rec {
...
+    # clang crash fix: https://github.com/dungeons-of-moria/umoria/pull/87
+    (fetchpatch {
+      name = "clang-crash.patch";
+      url = "https://github.com/dungeons-of-moria/umoria/commit/d073e8f867c49bb04a02c1995dd3efb0c5cc07e7.patch";
+      hash = "sha256-4uwO8fe4M5jt0IM0z6MjO8UaEezweMA5L+pusel4VUU=";
+    })
+  ];
+
+  postPatch = ''
+    # Do not apply blanter -Werror as it tends to fail on fresher
```
Comment (kenranunderscore) [inline pkgs/by-name/um/umoria/package.nix:52]: What's "blanter"? Is it a typo of "blanket"? :)

**Comment by kenranunderscore** [comment]
I've checked it out and shortly tried it with my old save game, so I can confirm that it works :+1:
As for the details, I'm not familiar enough with the matter, but not being restricted to `gcc` 9 seems nice.

File: pkgs/by-name/um/umoria/package.nix (hunk 2 above)
Comment (trofi) [reply pkgs/by-name/um/umoria/package.nix]: Yeah, good catch! Fixed to `blanket`.

**Review by kenranunderscore — APPROVED** [review]

**Review by kenranunderscore — APPROVED** [review]

**Pull request merged**